File: neetcode-250/two-pointers/1768-merge-strings-alternately.ts
function mergeAlternately(word1: string, word2: string): string {
	let i1 = 0,
		i2 = 0;
	let res: string[] = [];

	while (i1 < word1.length && i2 < word2.length) {
		res.push(word1[i1], word2[i2]);
		++i1;
		++i2;
	}

	if (i1 < word1.length) {
		res.push(word1.slice(i1));
	} else if (i2 < word2.length) {
		res.push(word2.slice(i2));
	}

	return res.join("");
}

File: neetcode-250/two-pointers/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res = []
        i = 0
        while i < min(len(word1), len(word2)):
            res.append(word1[i])
            res.append(word2[i])
            i += 1
        if i < len(word1):
            res.append(word1[i:])
        elif i < len(word2):
            res.append(word2[i:])
        return "".join(res)
